rust/src/http2/files.rs

```diff
@@ -0,0 +1,52 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use crate::core::*;
+use crate::filecontainer::*;
+
+/// Wrapper around Suricata's internal file container logic.
+#[derive(Debug)]
+//TODOask one generic structure for all parsers
```

Comment:
This `PROTOXFiles` structure is redundant with SMB and NFS at least.
Should we make one generic ?